classes/author/Collector.inc.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * @file classes/publication/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class publication
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
+ */
+
+namespace PKP\author;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\plugins\HookRegistry;
+
+class Collector implements CollectorInterface
+{
+    /** @var DAO */
+    public $dao;
+
+    /** @var array|null */
+    public $contextIds = null;
+
+    /** @var array|null */
+    public $submissionIds = null;
+
+    /** @var array|null */
+    public $publicationIds = null;
+
+    /** @var string get authors with a family name */
+    protected $familyName = '';
+
+    /** @var string get authors with a given name */
+    protected $givenName = '';
+
+    /** @var string get authors with a specified country code */
+    protected $country = '';
+
+    /** @var string get authors with a specified affiliation */
+    protected $affiliation = '';
+
+    /** @var int */
+    public $count = 0;
+
+    /** @var int */
+    public $offset = 0;
+
+    /** @var bool|null */
+    public $includeInBrowse = null;
+
+    /** @var bool */
+    public $orderByAuthorId = false;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Filter by contexts
+     */
+    public function filterByContextIds(array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    // /**
+    //  * Filter by submissions
+    //  */
+    // public function filterBySubmissionIds(array $submissionIds): self
+    // {
+    //     $this->submissionIds = $submissionIds;
+    //     return $this;
+    // }
+
+    /**
+     * Filter by publications
+     */
+    public function filterByPublicationIds(array $publicationIds): self
+    {
+        $this->publicationIds = $publicationIds;
+        return $this;
+    }
+
+    /**
+     * Filter by include in browse
+     */
+    public function filterByIncludeInBrowse(bool $includeInBrowse): self
+    {
+        $this->includeInBrowse = $includeInBrowse;
+        return $this;
+    }
+
+    /**
+     * Filter by include in browse
+     */
+    public function orderByAuthorId(bool $orderByAuthorId = false): self
+    {
+        $this->orderByAuthorId = $orderByAuthorId;
+        return $this;
+    }
```

Comment:
I'd prefer to make this a regular `orderBy` with a default value. So that would look something like

```php
class Collector implements CollectorInterface
{
    public const ORDERBY_SEQUENCE = 'sequence';
    public const ORDERBY_ID = 'id';

    public $orderBy = self::ORDERBY_SEQUENCE;

    public function orderBy(string $orderBy) : self
    {
        $this->orderBy = $orderBy;
        return $this;
    }
}
```

Then someone can set their own like:

```php
$collector->orderBy($collector::ORDERBY_ID);
```

Reply:
Nice!

Fixed